src/vs/workbench/services/search/node/rawSearchService.ts

```diff
@@ -313,29 +313,30 @@ export class SearchService implements IRawSearchService {
 		return new PPromise<ISerializedSearchComplete, ISerializedSearchProgressItem>((c, e, p) => {
 			// Use BatchedCollector to get new results to the frontend every 2s at least, until 50 results have been returned
 			const collector = new BatchedCollector<ISerializedFileMatch>(batchSize, p);
-			engine.search((matches) => {
-				const totalMatches = matches.reduce((acc, m) => acc + m.numMatches, 0);
-				collector.addItems(matches, totalMatches);
+			engine.search().then((success) => {
+				c(success);
+			}, (error) => {
+				e(error);
 			}, (progress) => {
-				p(progress);
-			}, (error, stats) => {
 				collector.flush();
-
-				if (error) {
-					e(error);
-				} else {
-					c(stats);
-				}
+				const results = progress.results;
```

Comment:
Which type of `const results`. Can you specify this type?

Reply:
Same question as below. I'm leaning towards _not_ specifying these implicit types everywhere.